api/types/backend/backend.go

```diff
@@ -98,6 +99,9 @@ type ExecProcessConfig struct {
 type ContainerCommitConfig struct {
 	types.ContainerCommitConfig
 	Changes []string
+	// TODO: ContainerConfig is only used by the dockerfile Builder, so remove it
+	// once the Builder has been updated to use a different interface
+	ContainerConfig *container.Config
```

Comment:
This is not actually part of the HTTP API (even though it's under `api/types`). It's only in the interface between the router and the daemon. I don't think we'll have to deprecate it, we can just remove it at some point.